runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -388,12 +394,63 @@ public void onGameTick(GameTick event)
 		lastPlayerLocation = client.getLocalPlayer().getWorldLocation();
 	}
 
+	private String buildRecoloredNpcMenuEntryTarget(MenuEntry menuEntry, NPC menuNpc)
+	{
+		List<Color> menuEntryColors = extractMenuEntryColors(menuEntry);
+
+		Color menuNpcNameColor = null;
+		if (config.highlightMenuNames() &&
+			highlightedNpcs.stream().anyMatch(npc -> menuNpc.getIndex() == npc.getIndex()))
+		{
+			menuNpcNameColor = config.getHighlightColor();
+		}
+		else
+		{
+			menuNpcNameColor = menuEntryColors.get(0);
+		}
+
+		if (config.dampenDeadNPCs() && menuNpc.isDead())
+		{
+			menuNpcNameColor = menuNpcNameColor.darker();
+		}
+		final String menuNpcNameColorHex = ColorUtil.colorToHexCode(menuNpcNameColor);
+
+		Color menuNpcLevelColor = null;
+		// NPCs without combat levels need to be treated differently.
+		if (menuNpc.getCombatLevel() > 0)
+		{
+			menuNpcLevelColor = menuEntryColors.get(1);
+			String menuNpcLevelColorHex = ColorUtil.colorToHexCode(menuNpcLevelColor);
+			return String.format(NPC_MENU_ENTRY_TARGET_FORMAT,
+				menuNpcNameColorHex, menuNpc.getName(),
+				menuNpcLevelColorHex, menuNpc.getCombatLevel());
+		}
+		else
+		{
+			return String.format(NPC_MENU_ENTRY_TARGET_NO_COMBAT_LVL_FORMAT,
+				menuNpcNameColorHex, menuNpc.getName());
+		}
+	}
+
 	private static boolean isInViewRange(WorldPoint wp1, WorldPoint wp2)
 	{
 		int distance = wp1.distanceTo(wp2);
 		return distance < MAX_ACTOR_VIEW_RANGE;
 	}
 
+	private static List<Color> extractMenuEntryColors(MenuEntry menuEntry)
+	{
+		Matcher matcher = COLOR_REGEX.matcher(menuEntry.getTarget());
+		List<Color> menuEntryColors = new ArrayList();
```

Comment:
Why does this need to be a list? You can use named capture groups or get the second group.

Reply:
The grouping order is different for menu entries like casting spells or using items on NPCs. I was thinking with #11595 in mind.

Reply:
You should add tests for this recolor option function.

Reply:
I've attempted to add some, can someone check I've used them correctly? I've never used java or RL unit tests before.